vllm_spyre/v1/worker/spyre_model_runner.py

```diff
@@ -110,6 +112,24 @@ def __init__(
         self._sampling_params_by_request: dict[str, SamplingParams] = {}
         self._max_logprobs: Optional[int] = None
 
+        self._req_ids2idx_prompt: dict = {}
+        self._req_ids2idx_decode: dict = {}
+        self._decode_batch_size = 0
+        self._active_pages = []
+        self._free_page_idxs = []
+        self._position_ids_prompt: torch.Tensor = None
+        self._mask_prompt: torch.Tensor = None
+        self._tkv: int = 0
+        self._tkv2fms: int = 0
+        self._prev_step_dec = False
```

Comment:
In my opinion, the management of pages here is implemented at the wrong level. We should not be trying to maintain all of this state in the model runner itself. We should be looking at how vLLM (V1) is implemented on GPU as a guide (e.g., we should be using something like the [InputBatch](https://github.com/vllm-project/vllm/blob/main/vllm/v1/worker/gpu_input_batch.py) class to maintain this state).  I think as a first attempt it is fine, and we can iteratively improve it from here.